### ISSUE TITLE
Header and Mega Menu Navigation

### ISSUE THREAD

**Issue by SteveJonesDev** [equalizedigital/student-loan-planner-theme#32]
- Header
  - site-title should be an H1 on the home page  
- Navigation
  - Keyboard users should be able to tab through parent menu items without having to tab through all the submenu links. See our website for an example. https://equalizedigital.com/
  - Mobile menu button
    - Aria attributes: `aria-controls="id-of-mobile-menu-container" aria-expanded="false"`
  -  Mobile menu
    - Tabbing exposes the 'back to all' button.
    - The submenu only seems to be opening for the first link and doesn't fill the page.
    - These are going to have to be treated like modals and lock focus. Use your keyboard when testing.
    - I can't access the search field with a keyboard
    - I think there are two search fields with the same ids. They should be unique. 
- Search
  - This needs to be treated like a modal.
  - Search button
    - Aria attributes: `aria-haspopup="dialog" aria-controls="id-of-popup" aria-expanded="false"`
  - Search popup
    - ID attribute matching the aria-controls attribute on the search button
    - Aria attributes: `role="dialog" aria-modal="true"`
    - Focus trap on the popup
    - When the close button is clicked focus returns back to the search button

Basecamp Ref: https://3.basecamp.com/3579237/buckets/32176113/card_tables/cards/6164734867
